src/views/Profile/components/EditProfileModal/ChangeProfilePicPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+import { Button, InjectedModalProps, Skeleton, Text } from '@pancakeswap-libs/uikit'
+import { useWallet } from '@binance-chain/bsc-use-wallet'
+import { useDispatch } from 'react-redux'
+import nftList from 'config/constants/nfts'
+import { useProfile, useToast } from 'state/hooks'
+import useI18n from 'hooks/useI18n'
+import { fetchProfile } from 'state/profile'
+import useGetWalletNfts from 'hooks/useGetWalletNfts'
+import useApproveConfirmTransaction from 'hooks/useApproveConfirmTransaction'
+import { usePancakeRabbits, useProfile as useProfileContract } from 'hooks/useContract'
+import { getPancakeProfileAddress, getPancakeRabbitsAddress } from 'utils/addressHelpers'
+import SelectionCard from '../SelectionCard'
+import ApproveConfirmButtons from '../ApproveConfirmButtons'
+
+type ChangeProfilePicPageProps = InjectedModalProps
+
+const ChangeProfilePicPage: React.FC<ChangeProfilePicPageProps> = ({ onDismiss }) => {
+  const [tokenId, setTokenId] = useState(null)
+  const TranslateString = useI18n()
+  const { isLoading, nfts: nftsInWallet } = useGetWalletNfts()
+  const dispatch = useDispatch()
+  const { profile } = useProfile()
+  const pancakeRabbitsContract = usePancakeRabbits()
+  const profileContract = useProfileContract()
+  const { account } = useWallet()
+  const { toastSuccess } = useToast()
+  const {
+    isApproving,
+    isApproved,
+    isConfirmed,
+    isConfirming,
+    handleApprove,
+    handleConfirm,
+  } = useApproveConfirmTransaction({
+    onApprove: () => {
+      return pancakeRabbitsContract.methods.approve(getPancakeProfileAddress(), tokenId).send({ from: account })
+    },
+    onConfirm: () => {
+      if (!profile.isActive) {
+        return profileContract.methods.reactivateProfile(getPancakeRabbitsAddress(), tokenId).send({ from: account })
+      }
+
+      return profileContract.methods.updateProfile(getPancakeRabbitsAddress(), tokenId).send({ from: account })
+    },
+    onSuccess: async () => {
+      // Re-fetch profile
+      await dispatch(fetchProfile(account))
```

Comment:
we probably don't need to wait here, you can just dispatch and show the message. Or at least show the message before the dispatch

Reply:
This is done because the modal closes and it doesn't update right away. This looks a lot better because when the modal closes the profile is updated by then.